packages/ingest/granule.js

```diff
@@ -50,45 +58,50 @@ class Discover {
   /**
    * Receives a file object and adds granule, bucket and path information
    * extracted from the collection record
-   * @param {object} file the file object
-   * @returns {object} Updated file with granuleId, bucket and path information
+   *
+   * @param {Object} file - the file object
+   * @returns {Object} - Updated file with granuleId, bucket and path information
    */
-  setGranuleInfo(_file) {
-    let granuleId;
-    const file = _file;
+  setGranuleInfo(file) {
+    const _file = file;
     let test = new RegExp(this.collection.granuleIdExtraction);
     const match = file.name.match(test);
+
     if (match) {
-      granuleId = match[1];
-      for (const f of this.collection.files) {
+      const granuleId = match[1];
+
+      this.collection.files.forEach((f) => {
         test = new RegExp(f.regex);
+        const urlPath = f.url_path || this.collection.url_path || '';
+
         if (file.name.match(test)) {
-          file.granuleId = granuleId;
-          file.bucket = this.buckets[f.bucket];
-          if (f.url_path) {
-            file.url_path = f.url_path;
-          }
-          else {
-            file.url_path = this.collection.url_path || '';
-          }
+          _file.granuleId = granuleId;
+          _file.bucket = this.buckets[f.bucket];
+          _file.url_path = urlPath;
         }
-      }
+      });
 
       // if collection regex matched, the following will be true
-      if (file.granuleId && file.bucket) {
-        return file;
+      if (_file.granuleId && _file.bucket) {
+        return _file;
       }
     }
+
     return false;
   }
 
+  /**
+  * Discover new files that match a given path
+  *
+  * @returns {Array} an array of granule objects
+  **/
   async discover() {
     // get list of files that matches a given path
     const updatedFiles = (await this.list())
       .map((file) => this.setGranuleInfo(file))
       .filter((file) => file);
 
-    return await this.findNewGranules(updatedFiles);
+    return this.findNewGranules(updatedFiles);
```

Comment:
Why did you remove the `await` here?

Reply:
`this.findNewGranules()` returns a Promise.  There is not need to wait for that promise to resolve before returning from the discover function.  Not having await here is correct.

Reply:
yeah, await is redundant when you are just passing a promise from another function